serializers/src/main/java/io/pravega/schemaregistry/codec/Codecs.java

```diff
@@ -0,0 +1,130 @@
+import org.xerial.snappy.Snappy;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * Utility class for creating codecs for none, snappy or gzip. 
+ */
+public enum Codecs {
+    None(Constants.NOOP),
+    GzipCompressor(Constants.GZIP_CODEC), 
+    SnappyCompressor(Constants.SNAPPY_CODEC);
+
+    @Getter
+    private final Codec codec;
+    
+    Codecs(Codec codec) {
+        this.codec = codec;  
+    }
+
+    private static class Noop implements Codec {
+        private static final CodecType CODEC_TYPE_NONE = new CodecType(Constants.NONE);
+
+        @Override
+        public String getName() {
+            return CODEC_TYPE_NONE.getName();
+        }
+
+        @Override
+        public CodecType getCodecType() {
+            return CODEC_TYPE_NONE;
+        }
+
+        @Override
+        public void encode(ByteBuffer data, ByteArrayOutputStream bos) {
+        }
+
+        @Override
+        public ByteBuffer decode(ByteBuffer data, Map<String, String> codecProperties) {
+            return data;
+        }
+    }
+
+    private static class GZipCodec implements Codec {
+        private static final CodecType CODEC_TYPE_GZIP = new CodecType(Constants.APPLICATION_X_GZIP);
+        @Override
+        public String getName() {
+            return CODEC_TYPE_GZIP.getName();
+        }
+
+        @Override
+        public CodecType getCodecType() {
+            return CODEC_TYPE_GZIP;
+        }
+
+        @Override
+        public void encode(ByteBuffer data, ByteArrayOutputStream bos) throws IOException {
+            try (GZIPOutputStream gzipOS = new GZIPOutputStream(bos)) {
+                gzipOS.write(data.array(), data.arrayOffset() + data.position(), data.remaining());
+            }
+        }
+
+        @Override
+        public ByteBuffer decode(ByteBuffer data, Map<String, String> codecProperties) throws IOException {
+            InputStream bis = new ByteBufferBackedInputStream(data);
+            return ByteBuffer.wrap(IOUtils.toByteArray(new GZIPInputStream(bis)));
+        }
+    }
+
+    private static class SnappyCodec implements Codec {
+        private static final CodecType CODEC_TYPE_SNAPPY = new CodecType(Constants.APPLICATION_X_SNAPPY_FRAMED);
+        @Override
+        public String getName() {
+            return CODEC_TYPE_SNAPPY.getName();
+        }
+
+        @Override
+        public CodecType getCodecType() {
+            return CODEC_TYPE_SNAPPY;
+        }
+
+        @Override
+        public void encode(ByteBuffer data, ByteArrayOutputStream bos) throws IOException {
+            int capacity = Snappy.maxCompressedLength(data.remaining());
+            byte[] encoded = new byte[capacity];
+            
+            int size = Snappy.compress(data.array(), data.arrayOffset() + data.position(),
```

Comment:
I am not entirely sure how to handle the case when bytebuffer does not have the backing data. 

At least for codecs, the bytebuffer we pass will have the backing data because the serializer creates a bytebuffer backed by an array of serialized data.